ArduPlane/GCS_Mavlink.cpp

```diff
@@ -2103,6 +2103,39 @@ void GCS_MAVLINK_Plane::handleMessage(mavlink_message_t* msg)
         plane.DataFlash.remote_log_block_status_msg(chan, msg);
         break;
 
+    // BEGIN GTRI MOD ////////////
+    case MAVLINK_MSG_ID_SET_ATTITUDE_TARGET:
+    {
+        //Only allow companion computer (or other external controller) to 
+        //control attitude in GUIDED mode.  We DON'T want external control
+        //in e.g., RTL, CICLE. Specifying a single mode for companion 
+        //computer control is more safe (even more so when using
+        //FENCE_ACTION = 4 for geofence failures).
+        if (plane.control_mode != GUIDED) { //don't screw up failsafes
+            break; 
+        }
+
+        mavlink_set_attitude_target_t att_target;
+        mavlink_msg_set_attitude_target_decode(msg, &att_target);
+        //Unexepectedly, the mask is expecting "ones" for dimensions that should
+        //be IGNORNED rather than INCLUDED.  See mavlink documentation of the
+        //SET_ATTITUDE_TARGET message, type_mask field. 
+        const uint16_t roll_mask = 0b11111110; // (roll mask at bit 1)
+
+        if (att_target.type_mask & roll_mask) {
+            // Set the flag for external roll to the nav control
+            plane.guiding_roll = true;
```

Comment:
what if you only get a single packet with this roll type? Seems the rolling flag is set forevever until you switch modes. As @squilter said, this needs a timeout. Something for the guided_state!